src/argilla/client/sdk/users/models.py

```diff
@@ -12,17 +12,35 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from datetime import datetime
 from typing import List, Optional
+from uuid import UUID
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
 
+from argilla.server.models import UserRole as Roles
 
-class User(BaseModel):
-    """User data model"""
 
+class UserCreateModel(BaseModel):
+    first_name: str = Field(min_length=1)
+    last_name: Optional[str] = Field(min_length=1)
+    username: str = Field(min_length=1, regex=r"^(?!-|_)[a-z0-9-_]+$")
+    role: Roles = "annotator"
```

Comment:
```suggestion
    role: Roles = Roles.annotator
```

Reply:
Regarding how `Enum`s are accessed in Python 3.11, do you know if this may eventually lead to any issue? cc @gabrielmbmb